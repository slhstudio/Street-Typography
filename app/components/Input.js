import React from 'react';


const Input = (props) => {
    return (
      <div>
        <input
          id={props.id}
          type={props.type}
          name={props.name}
          placeholder={props.placeholder}
          value={props.value}
          onChange={props.handleChange}
        />
      </div>
    )
}



export default Input;














