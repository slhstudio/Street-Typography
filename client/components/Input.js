import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledInput = styled.input`
  border: .2em solid black;
  padding: 2em;
  margin: 1em 20% 1em 20%;
`;

const Input = (props) => {
  return (
    <StyledInput
      id={props.name}
      type={props.type}
      name={props.name}
      accept={props.accept}
      placeholder={props.placeholder}
      value={props.value}
      onChange={props.handleChange}
      required={props.required}
    />
  )
}

Input.propTypes = {
  type: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  accept: PropTypes.string,
  placeholder: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  handleChange: PropTypes.func.isRequired
}



export default Input;














