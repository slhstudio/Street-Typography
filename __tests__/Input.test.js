import React from 'react';
import Input from '../client/components/Input';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeProps = {
  type : 'text',
  name : 'notes',
  placeholder : 'Add some notes here',
  value : 'neon!',
}

const newProps = {
  type : 'text',
  name : 'address',
  placeholder : 'Add address',
  value : 'Portland, Maine'
}

describe('<Input/>', () => {
  it ('matches snapshot', () => {
    const wrapper = shallow(<Input {...fakeProps}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it ('updates controlled input values correctly', () => {
    const wrapper = shallow(<Input {...fakeProps}/>);
    wrapper.setProps({...newProps});
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});