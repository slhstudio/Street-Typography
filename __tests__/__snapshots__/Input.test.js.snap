// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> matches snapshot 1`] = `
<styled.input
  id="notes"
  name="notes"
  onChange={[MockFunction]}
  placeholder="Add some notes here"
  type="text"
  value="neon!"
/>
`;

exports[`<Input/> updates controlled input values correctly 1`] = `
<styled.input
  id="address"
  name="address"
  onChange={[MockFunction]}
  placeholder="Add address"
  type="text"
  value="Portland, Maine"
/>
`;
